Python/emscripten_trampoline.c

```diff
@@ -1,92 +1,220 @@
+      0x01, // 1 import (table $funcs (import "e" "t") 0 funcref)
+      0x01, 0x65, // "e"
+      0x01, 0x74, // "t"
+      0x01,       // importing a table
+      0x70,       // of entry type funcref
+      0x00, 0x00, // table limits: no max, min of 0
+    0x03, 0x02,   // Function section
+      0x01, 0x01, // We're going to define one function of type 1 (func (param i32) (result i32))
+    0x07, 0x05, // export section
+      0x01, // 1 export
+        0x01, 0x66, // called "f"
+        0x00, // a function
+        0x00, // at index 0
+
+    0x0a, 0x52,  // Code section,
+      0x01, 0x50, // one entry of length 50
+      0x01, 0x01, 0x70, // one local of type funcref
+      // Body of the function
+      0x20, 0x00,       // local.get $fptr
+      0x25, 0x00,       // table.get $funcs
+      0x22, 0x01,       // local.tee $fref
+      0xfb, 0x14, 0x04, // ref.test $type4
+      0x02, 0x05,       // block $b (type $blocktype)
+        0x45,           //   i32.eqz
+        0x0d, 0x00,     //   br_if $b
+        0x41, 0x04,     //   i32.const 4
+        0x0f,           //   return
+      0x0b,             // end block
+
+      0x20, 0x01,       // local.get $fref
+      0xfb, 0x14, 0x03, // ref.test $type3
+      0x02, 0x05,       // block $b (type $blocktype)
+        0x45,           //   i32.eqz
+        0x0d, 0x00,     //   br_if $b
+        0x41, 0x03,     //   i32.const 3
+        0x0f,           //   return
+      0x0b,             // end block
+
+      0x20, 0x01,       // local.get $fref
+      0xfb, 0x14, 0x02, // ref.test $type2
+      0x02, 0x05,       // block $b (type $blocktype)
+        0x45,           //   i32.eqz
+        0x0d, 0x00,     //   br_if $b
+        0x41, 0x02,     //   i32.const 2
+        0x0f,           //   return
+      0x0b,             // end block
+
+      0x20, 0x01,       // local.get $fref
+      0xfb, 0x14, 0x01, // ref.test $type1
+      0x02, 0x05,       // block $b (type $blocktype)
+        0x45,           //   i32.eqz
+        0x0d, 0x00,     //   br_if $b
+        0x41, 0x01,     //   i32.const 1
+        0x0f,           //   return
+      0x0b,             // end block
 
+      0x20, 0x01,       // local.get $fref
+      0xfb, 0x14, 0x00, // ref.test $type0
+      0x02, 0x05,       // block $b (type $blocktype)
+        0x45,           //   i32.eqz
+        0x0d, 0x00,     //   br_if $b
+        0x41, 0x00,     //   i32.const 0
+        0x0f,           //   return
+      0x0b,             // end block
+
+      0x41, 0x7f,       // i32.const -1
+      0x0b // end function
+  ]);
+  let ptr = 0;
+  try {
+    const mod = new WebAssembly.Module(code);
+    const inst = new WebAssembly.Instance(mod, {e: {t: wasmTable}});
+    ptr = addFunction(inst.exports.f);
+  } catch(e) {
+    // If something goes wrong, we'll null out _PyEM_CountFuncParams and fall
+    // back to the JS trampoline.
+  }
+  HEAP32[__PyRuntime/4 + EMSCRIPTEN_COUNT_ARGS_OFFSET] = ptr;
```

Comment:
Hmm we seem to set this too early and written over in pystate. Either we need to use a separate global, or we need to cache the value in JS and _also_ write it in pystate.